### PR TITLE
Add label smooth support for dygraph

### PR DESCRIPTION
**Pull request by littletomatodonkey**
1. Fix model inferface for dpn, densenet and hrnet.
2. Add label smooth support for dygraph.

### CODE REVIEW

File: tools/program.py [hunk 1]
```diff
@@ -297,9 +295,11 @@ def compute(config, out, label, mode='train'):
         dataloader(): a bridge between the model and the data
         fetchs(dict): dict of model outputs(included loss and measures)
     """
+    out = net(feeds["image"])
     fetchs = create_fetchs(
+        feeds,
         out,
-        label,
+        config,
```
Comment (shippingwang) [inline tools/program.py:302]: It seems that passing ```config``` is enough.

File: tools/program.py [hunk 2]
```diff
@@ -51,9 +51,7 @@ def create_dataloader():
     trainer_num = int(os.environ.get('PADDLE_TRAINERS_NUM', 1))
```
Comment (shippingwang) [inline tools/program.py:51]: 这里其实不存在 0 的情况了吧 我们可以改成 == 1

File: tools/program.py [hunk 3]
```diff
@@ -324,19 +338,30 @@ def run(dataloader, config, net, optimizer=None, epoch=0, mode='train'):
...
+        ])
 
     tic = time.time()
-    for idx, (img, label) in enumerate(dataloader()):
-        label = to_variable(label.numpy().astype('int64').reshape(-1, 1))
-        fetchs = compute(config, net(img), label, mode)
+    for idx, batch in enumerate(dataloader()):
+        bs = len(batch[0])
+        feeds = create_feeds(batch, use_mix)
+        fetchs = compute(feeds, net, config, mode)
```
Comment (shippingwang) [inline tools/program.py:364]: 动态图直接用create_fetchs会清楚点

File: tools/program.py [hunk 4]
```diff
@@ -310,6 +310,20 @@ def compute(config, out, label, mode='train'):
...
 
+def create_feeds(batch, use_mix):
+    if use_mix:
+        image = batch[0]
+        y_a = to_variable(batch[1].numpy().astype("int64").reshape(-1, 1))
+        y_b = to_variable(batch[2].numpy().astype("int64").reshape(-1, 1))
+        lam = to_variable(batch[3].numpy().astype("float32").reshape(-1, 1))
+        feeds = {"image": image, "y_a": y_a, "y_b": y_b, "lam": lam}
+    else:
+        image = batch[0]
```
Comment (dyning) [inline tools/program.py:321]: image也转换为to_variable

File: tools/program.py [hunk 5]
```diff
@@ -324,19 +338,30 @@ def run(dataloader, config, net, optimizer=None, epoch=0, mode='train'):
...
+                'lr', 'f', need_avg=False)),
+            ("batch_time", AverageMeter('elapse', '.3f')),
+        ])
 
     tic = time.time()
-    for idx, (img, label) in enumerate(dataloader()):
-        label = to_variable(label.numpy().astype('int64').reshape(-1, 1))
-        fetchs = compute(config, net(img), label, mode)
+    for idx, batch in enumerate(dataloader()):
+        bs = len(batch[0])
```
Comment (dyning) [inline tools/program.py:362]: bs改写为batch_size

File: tools/program.py (hunk 2 above)
Comment (littletomatodonkey) [reply tools/program.py:51]: done

File: tools/program.py (hunk 1 above)
Comment (littletomatodonkey) [reply tools/program.py]: done

File: tools/program.py (hunk 3 above)
Comment (littletomatodonkey) [reply tools/program.py]: done

File: tools/program.py (hunk 5 above)
Comment (littletomatodonkey) [reply tools/program.py]: done

File: tools/program.py (hunk 4 above)
Comment (littletomatodonkey) [reply tools/program.py]: done

File: tools/program.py [hunk 6]
```diff
@@ -278,36 +277,17 @@ def mixed_precision_optimizer(config, optimizer):
...
-        label,
-        config.ARCHITECTURE,
-        config.topk,
-        config.classes_num,
-        epsilon=config.get('ls_epsilon'),
-        use_mix=config.get('use_mix') and mode == 'train',
-        use_distillation=config.get('use_distillation'))
-
-    return fetchs
+def create_feeds(batch, use_mix):
```
Comment (WuHaobo) [inline tools/program.py:280]: 这个函数注释可以补一下

Comment (littletomatodonkey) [reply tools/program.py:280]: 好的，下个PR补充

**Pull request merged**